### PR TITLE
Fix deployment wizard to add the `WAZUH_PROTOCOL` variable only for UDP connections

### PR DESCRIPTION
**Pull request by chantal-kelm**
### Description

Now, the `WAZUH_PROTOCOL` variable is only displayed if a remote [connection](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/remote.html#connection) is of type **secure** and its protocol is **UDP**.

**Notice**

> This issue was developed using Imposter, as we have not yet received a response from the @wazuh/framework team of a possible bug in the API. This issue must not be merged yet, as the code will be updated as soon as we have a reply from the @wazuh/framework team.
 
### Issues Resolved
https://github.com/wazuh/wazuh-kibana-app/issues/4318

### Evidence

Connection: Secure Protocol: UDP :

https://user-images.githubusercontent.com/99441266/198362522-3a11be0f-5339-4fe1-a80f-ab74ea004261.mp4

Connection: Secure Protocol: TCP :

https://user-images.githubusercontent.com/99441266/198362710-14078974-98da-4f83-bf17-723f40dee910.mp4

Connection: Syslog

https://user-images.githubusercontent.com/99441266/198362818-e2f8f5e2-e349-4427-825c-2256fee6f7f1.mp4

Error handling and warning in the UI to the client.

https://user-images.githubusercontent.com/99441266/198363223-246b9ddf-53b3-42cf-98f2-bd28a97cc32e.mp4

### Test
*Go to the Agents' tab
*Click on the 'Deploy new agent' button
*Verify that when a connection is type secure and its protocol is UDP, the variable WAZUH_PROTOCOL appears in the installation command, otherwise it does not appear.

You can use this data to test with the imposter:
CASE 1: Variable `WAZUH_PROTOCOL` should appear in the installation command.

```json
{
    "data": {
      "remote": [
        {
          "connection": "syslog",
          "ipv6": "no",
          "protocol": [
            "UDP"
          ],
          "port": "514",
          "allowed-ips": [
            "0.0.0.0/0"
          ]
        },
        {
          "connection": "secure",
          "ipv6": "no",
          "protocol": [
            "UDP"
          ],
          "port": "1514",
          "queue_size": "131072"
        }
      ]
    },
    }, "error": 0
  }
```

Case 2: WAZUH_PROTOCOL variable should not appear in the install command

```
{
    "data": {
      "remote": [
        {
          "connection": "syslog",
          "ipv6": "no",
          "protocol": [
            "UDP"
          ],
          "port": "514",
          "allowed-ips": [
            "0.0.0.0/0"
          ]
        },
        {
          "connection": "secure",
          "ipv6": "no",
          "protocol": [
            "TCP"
          ],
          "port": "1514",
          "queue_size": "131072"
        }
      ]
    },
    }, "error": 0
  }
```

### CODE REVIEW

File: public/controllers/agent/components/register-agent.js [hunk 1]
```diff
@@ -221,8 +221,19 @@ export const RegisterAgent = withErrorBoundary(
     async getRemoteInfo() {
       try {
         const result = await WzRequest.apiReq('GET', '/agents/000/config/request/remote', {});
+        console.log(result, 'resultt+++')
+        console.log('remoteee+')
         const remote = ((result.data || {}).data || {}).remote || {};
-        return (remote[0] || {}).protocol !== 'tcp' && (remote[0] || {}).protocol[0] !== 'TCP';
+        const remoteFiltered = remote.filter((item, key) => {
+          if (item.connection === 'secure') {
+            if(item.protocol.length === 1 && item.protocol[0] == 'UDP') {
```
Comment (Desvelao) [inline public/controllers/agent/components/register-agent.js:229]: **nitpick**: The `protocol` property is an array that can contain `'TCP'`, `'UDP'` or both. On this line, we are trusting the `'UDP'` value will be at the 0 index. From my tests, configurating the manager as:
```xml
<remote>
    <connection>secure</connection>
    <port>1514</port>
    <protocol>UDP,tcp</protocol>
    <queue_size>131072</queue_size>
  </remote>
```

The API response to `/agents/000/config/request/remote`:
```json
{
  "data": {
    "remote": [
      {
        "connection": "secure",
        "ipv6": "no",
        "protocol": [
          "TCP",
          "UDP"
        ],
        "port": "1514",
        "queue_size": "131072"
      }
    ]
  },
  "error": 0
}
```

Note the order is `'TCP'` and `'UDP'` that is different of the configuration. This means the API response sorts the protocols. I would avoid checking the 0 index to know what protocol to use and see another way. The current solution could cause bugs if the API response is not sorted as it does at this moment.

References: https://github.com/wazuh/wazuh-kibana-app/issues/4318#issuecomment-1261081676

File: public/controllers/agent/components/register-agent.js [hunk 2]
```diff
@@ -221,8 +221,19 @@ export const RegisterAgent = withErrorBoundary(
     async getRemoteInfo() {
       try {
         const result = await WzRequest.apiReq('GET', '/agents/000/config/request/remote', {});
```
Comment (Desvelao) [inline public/controllers/agent/components/register-agent.js:223]: **nitpick**: If I am not wrong, this request gets the data of the `master` node. Related to this pull request https://github.com/wazuh/wazuh-kibana-app/pull/4737 that adds the selection of the manager to connect, the protocol should use the configurated by the selected manager.

For another hand, what should be the value of the protocol variable for the next situation (taking into account the mentioned pull request):
- manager1: tcp, udp
- manager2: udp

**Review by yenienserrano — DISMISSED** [review]
CR: ☑️ with comments
Test: ✅

**Review by Mayons95 — APPROVED** [review]
TR: :heavy_check_mark: 
LGTM


Test details

<details>

Screenshot from 2022-10-28 08-08-42

Screenshot from 2022-10-28 08-07-46

Screenshot from 2022-10-28 08-06-26

Screenshot from 2022-10-28 08-01-12

Screenshot from 2022-10-28 07-55-24


</details>

File: public/controllers/agent/components/register-agent.js (hunk 2 above)
Comment (Machi3mfl) [reply public/controllers/agent/components/register-agent.js]: Added comment in the linked issue to tread this topic: [https://github.com/wazuh/wazuh-kibana-app/issues/4318#issuecomment-1295303570](https://github.com/wazuh/wazuh-kibana-app/issues/4318#issuecomment-1295303570)

**Comment by AlexRuiz7** [comment]
Superseded by https://github.com/wazuh/wazuh-kibana-app/pull/4776

**Pull request closed without merging**